cmd/bootstrap/run/execution_state.go

```diff
@@ -67,11 +55,30 @@ func GenerateExecutionState(
 		<-compactor.Done()
 	}()
 
-	return bootstrap.NewBootstrapper(
-		zerolog.Nop()).BootstrapLedger(
-		ledgerStorage,
-		accountKey,
-		chain,
-		bootstrapOptions...,
-	)
+	stateCommitment, err := bootstrap.
+		NewBootstrapper(zerolog.Nop()).
+		BootstrapLedger(
+			ledgerStorage,
+			accountKey,
+			chain,
+			bootstrapOptions...,
+		)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+
+	matchTrie, err := ledgerStorage.FindTrieByStateCommit(stateCommitment)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+	if matchTrie == nil {
+		return flow.DummyStateCommitment, fmt.Errorf("bootstraping failed to produce a checkpoint")
```

Comment:
better to log the state commitment
```suggestion
		return flow.DummyStateCommitment, fmt.Errorf("bootstraping failed to produce a checkpoint for trie %v", stateCommitment)
```